stdlib/public/core/Flatten.swift.gyb

```diff
@@ -313,40 +270,50 @@ public struct ${Collection}<
   /// reachable from `startIndex` by zero or more applications of
   /// `successor()`.
   public var endIndex: Index {
-    return ${Index}(_endIndexOfFlattened: _base)
+    return ${Index}(_base.endIndex, nil)
   }
```

Comment:
Oh, I don't think that we want to add more storage to our lazy views for such a cached property.

We in general prefer `var { get }` over `let` because `let` can require storage to cache, and it gives more guarantees than `var`, which is important for library evolution.  We can't replace a `let` with a `var { get }` in future.